iroh-gateway/src/core.rs

```diff
@@ -0,0 +1,412 @@
+use axum::{
+    body::{self, Body, BoxBody},
+    error_handling::HandleErrorLayer,
+    extract::{Extension, Path, Query},
+    http::{header::*, StatusCode},
+    response::IntoResponse,
+    routing::get,
+    BoxError, Router,
+};
+use serde::Deserialize;
+use std::{borrow::Cow, collections::HashMap, time::Duration};
+use tower::ServiceBuilder;
+
+use crate::{
+    client::{Client, Request},
+    config::Config,
+    constants::*,
+    error::GatewayError,
+    response::{GatewayResponse, ResponseFormat},
+};
+
+pub struct Core {
```

Comment:
should derive `Debug` to make everyones lives easier down the road (similar for other structs)